packages/svelte2tsx/src/htmlxtojsx_v2/utils/node-utils.ts

```diff
@@ -0,0 +1,117 @@
+import { Node } from 'estree-walker';
+import MagicString from 'magic-string';
+import { BaseDirective } from '../../interfaces';
+
+export type TransformationArray = Array<string | [number, number]>;
+
+/**
+ * Moves or inserts text to a specific location in order.
+ * "In order" means that the transformation of the text before
+ * the given position reads exactly what was moved/inserted
+ * from left to right.
+ * After the transformation is done, everything inside the start-end-range that was
+ * not moved will be removed.
+ */
+export function transform(
```

Comment:
This is the heart of the new approach. This function ensure that we can write code that is better understandable by creating an array of transforms. They are passed to this function which then moves/appends text in order to the end position and then removes any code that was not moved, freeing us from having to cleanup stuff like `<` or `{` manually.